Orbit.h

```diff
@@ -1,122 +1,104 @@
-#define ORBIT_SEND(ptr, num) OrbitSendData(ptr, num)
-
-//-----------------------------------------------------------------------------
-struct OrbitAPI {
-  static inline bool Connect(const char* a_Host, int a_Port = 1789);
-  static inline bool IsConnected();
-
- private:
-  static inline void Init();
-
-  typedef void (*InitRemote)(char*);
-  typedef bool (*GetBool)();
-
-  static inline HINSTANCE& GetHandle();
-  static inline InitRemote& GetInitRemote();
-  static inline GetBool& GetIsConnected();
-};
+// While the main feature of Orbit is its ability to dynamically instrument
+// functions, manual instrumentation is still possible using the macros below.
+// These macros call empty functions that Orbit dynamically instruments.
 
-//-----------------------------------------------------------------------------
-#define ORBIT_NOOP       \
-  static volatile int x; \
-  x;
-#define ORBIT_LITERAL(x) ("" x)
+// To disable manual instrumentation macros, set ORBIT_API_ENABLED to 0.
+#define ORBIT_API_ENABLED 1
 
-//-----------------------------------------------------------------------------
-#pragma optimize( \
-    "", off)  // On purpose to prevent compiler from stripping functions
-// NOTE: Do not use these directly, use above macros instead
-__declspec(noinline) inline void OrbitStart(const char*) { ORBIT_NOOP; }
-__declspec(noinline) inline void OrbitStop() { ORBIT_NOOP; }
-__declspec(noinline) inline void OrbitLog(const char*) { ORBIT_NOOP; }
-__declspec(noinline) inline void OrbitSendData(void*, int) { ORBIT_NOOP; }
-#pragma optimize("", on)
-
-//-----------------------------------------------------------------------------
-struct OrbitScope {
-  OrbitScope(const char* a_Name) { OrbitStart(a_Name); }
-  ~OrbitScope() { OrbitStop(); }
-};
+#if ORBIT_API_ENABLED
+
+// ORBIT_SCOPE: profile current scope.
+#define ORBIT_SCOPE(name) orbit::Scope ORBIT_UNIQUE(ORB)(ORBIT_LITERAL(name))
+
+// ORBIT_START/ORBIT_STOP: profile time span on a single thread.
+#define ORBIT_START(name) orbit::Start(ORBIT_LITERAL(name))
+#define ORBIT_STOP orbit::Stop()
+
+// ORBIT_START_ASYNC/ORBIT_STOP_ASYNC: profile time span across threads.
+#define ORBIT_START_ASYNC(name, id) orbit::StartAsync(ORBIT_LITERAL(name), id)
+#define ORBIT_STOP_ASYNC(id) orbit::StopAsync(id)
+
+// ORBIT_[type]: graph variables.
+#define ORBIT_INT(name, value) orbit::TrackInt(ORBIT_LITERAL(name), value)
+#define ORBIT_INT64(name, value) orbit::TrackInt64(ORBIT_LITERAL(name), value)
+#define ORBIT_UINT(name, value) orbit::TrackUint(ORBIT_LITERAL(name), value)
+#define ORBIT_UINT64(name, value) orbit::TrackUint64(ORBIT_LITERAL(name), value)
+#define ORBIT_FLOAT(name, value) orbit::TrackFloat(ORBIT_LITERAL(name), value)
+#define ORBIT_DOUBLE(name, value) orbit::TrackDouble(ORBIT_LITERAL(name), value)
 
-//-----------------------------------------------------------------------------
+#else
+
+#define ORBIT_SCOPE(name)
+#define ORBIT_START(name)
+#define ORBIT_STOP
```

Comment:
Shouldn't this look like a function call like all the other ones?

Reply:
Yes, done.